tests/deploy/test_scheduler_agent.py

```diff
@@ -1232,6 +1215,67 @@ def make_resources(
     assert agent.executor_manager.executors["agent2"].execute_count == 3
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
+    #########################
+    # Verify failure events #
+    #########################
+    agent.executor_manager.reset_executor_counters()
+    # make agent1's executor also managed to make r1 fail and succeed as we please
+    executor1: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent1", executor1)
+
+    # release change to r1, and make its deploy fail
+    resources = make_resources(r1_value=2, r2_value=1, r3_value=0)
+    await agent.scheduler._new_version(18, resources, make_requires(resources))
+    await retry_limited_fast(lambda: rid1 in executor1.deploys)
+    executor1.deploys[rid1].set_result(const.HandlerResourceState.failed)
+    # assert that r2 got skipped for its dependencies through event propagation
```

Comment:
```suggestion
    # assert that r2 got scheduled for deploy through event propagation
```
We don't assert that it skips, we *make* it skip. At this point, the part we care about is simply that it gets the event.